arm/Microsoft.ContainerRegistry/registries/deploy.bicep

```diff
@@ -208,9 +217,12 @@ resource registry 'Microsoft.ContainerRegistry/registries@2021-09-01' = {
   }
   properties: {
     adminUserEnabled: acrAdminUserEnabled
-    encryption: acrSku == 'Premium' ? {
-      keyVaultProperties: !empty(keyVaultProperties) ? keyVaultProperties : null
-      status: encryptionStatus
+    encryption: !empty(cMKKeyName) ? {
```

Comment:
Should we keep the acrSku == 'Premium' condition? probably something to align on all other CMK implementations when relevant. Also worth mentioning in the cMK input parameter descriptions

Reply:
Yes and no. It is a good question that I've also seen other places. The question is, whether we want users to fail with (hopefully a meaningful error message), or shield them from that, but potentially hide (in this instance) that the encryption was actually not deployed. Thoughts?